File: date-time-ui/src/shared/UserDTO.ts
export class UserDTO{

  userId : number;
  name : string;
  dateOfCreation : Date;
  timeOfCreation : string;

  constructor(userId: number, name : string, dateOfCreation : Date, timeOfCreation : string){

    this.userId = userId;
    this.name = name;
    this.dateOfCreation = dateOfCreation;
    this.timeOfCreation = timeOfCreation;
  }
}

File: date-time-format/src/main/java/com/datetime/format/dto/UserDTO.java
package com.datetime.format.dto;

import java.time.LocalDateTime;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {

	private long userId;
	
	private String name;
	
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
	private Date dateOfCreation;
	
	private LocalDateTime timeOfCreation;
}
